lib/bolt/cli/src/commands/admin/cluster/datacenter.rs

```diff
@@ -0,0 +1,142 @@
+use anyhow::*;
+use bolt_core::context::ProjectContext;
+use clap::{Parser, ValueEnum};
+use rivet_api::{
+	apis::{configuration::Configuration, *},
+	models,
+};
+
+#[derive(ValueEnum, Clone)]
+pub enum DatacenterAdminProvider {
+	Linode,
+}
+
+impl From<DatacenterAdminProvider> for models::AdminProvider {
+	fn from(provider: DatacenterAdminProvider) -> Self {
+		match provider {
+			DatacenterAdminProvider::Linode => models::AdminProvider::Linode,
+		}
+	}
+}
+
+#[derive(ValueEnum, Clone)]
+pub enum DatacenterAdminBuildDeliveryMethod {
+	TrafficServer,
+}
+
+impl From<DatacenterAdminBuildDeliveryMethod> for models::AdminBuildDeliveryMethod {
+	fn from(method: DatacenterAdminBuildDeliveryMethod) -> Self {
+		match method {
+			DatacenterAdminBuildDeliveryMethod::TrafficServer => {
+				models::AdminBuildDeliveryMethod::TrafficServer
+			}
+		}
+	}
+}
+
+#[derive(Parser)]
+pub enum SubCommand {
+	/// Creates a new datacenter
+	Create {
+		/// The name id of the datacenter
+		#[clap(long)]
+		name_id: String,
+		/// The display name of the datacenter
+		#[clap(long)]
+		display_name: String,
+		/// The provider of the datacenter
+		#[clap(long)]
+		provider: DatacenterAdminProvider,
+		/// The provider datacenter ID
+		#[clap(long)]
+		provider_datacenter_id: String,
+		/// The build delivery method
+		#[clap(long)]
+		build_delivery_method: DatacenterAdminBuildDeliveryMethod,
+	},
+	/// Deletes an existing datacenter
+	Delete,
+	Taint,
+}
+
+impl SubCommand {
+	pub async fn execute(
+		self,
+		ctx: ProjectContext,
+		_config: Configuration,
+		cluster_name_id: String,
+		datacenter_name_id: Option<String>,
+	) -> Result<()> {
+		let api_admin_token = ctx.read_secret(&["rivet", "api_admin", "token"]).await?;
+
+		let config = Configuration {
+			base_path: ctx.origin_api(),
+			bearer_access_token: Some(api_admin_token),
+			..Default::default()
+		};
+
+		match (self, datacenter_name_id) {
+			(
+				Self::Create {
+					name_id,
+					display_name,
+					provider,
+					provider_datacenter_id,
+					build_delivery_method,
+				},
+				None,
+			) => {
+				admin_cluster_datacenter_api::admin_cluster_datacenter_create(
+					&config,
+					&cluster_name_id,
+					models::AdminClusterDatacenterCreateRequest {
+						name_id,
+						display_name,
+						provider: provider.into(),
+						provider_datacenter_id,
+						build_delivery_method: build_delivery_method.into(),
+						// Default to 4 hours
```

Comment:
are able to use the `duration` mod from rivet-util?

Reply:
ooh this is dope, will do :+1: